homework/list.js

```diff
@@ -50,30 +57,35 @@
       // Note: buff is captured, so we now have a
       // little closure naturally.
       buff.appendChild(li);
-    });
+    }).bind(this));
     ul.appendChild(buff);
     _wrapper.appendChild(ul);
   }
 
-  function fetchList(afterFetch) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
-    xhr.responseType = 'json';
-    xhr.onreadystatechange = function(e) {
-      // Watch out: we have a mysterious unknown 'this'.
-      if (this.readyState === 4 && this.status === 200) {
-        var listData = this.response;
-        // The flow ends here.
-        afterFetch(listData);
-      } else if (this.status !== 200 ){
-        // Ignore error in this case.
-      }
-    };
-    xhr.send();
+  list.prototype.fetchList = function(afterfetch) {
```

Comment:
Since you have successfully "Promisified" the method, you should remove the callback argument.